code/modules/chemistry/Chemistry-Recipes.dm

```diff
@@ -4333,6 +4333,56 @@
 		on_reaction(datum/reagents/holder, created_volume) // TODO: actual byproduct/multi-output handling
 			holder.add_reagent("phenol", created_volume, temp_new = holder.total_temperature, chemical_reaction = TRUE)
 
+	espresso //makin' caffeine by dehydrating coffee
+		name = "Coffee concentration"
+		id = "espresso"
+		result = "espresso" 
+		eventual_result = "espresso"
+		required_reagents = list("coffee" = 2, "sodium_sulfate" = 0.1) //sodium sulfate as an approximate drying agent
+		result_amount = 1
+		max_temperature = T0C + 30 //sodium sulfate fails at 30c (A.K.A. 'you want to to make the caffeine in another thing')
+		mix_phrase = "A gross precipitate forms as the water is absorbed."
+		instant = FALSE
+		reaction_speed = 0.5
+		var/shock_ticks = 0
+
+		physical_shock(var/force, var/datum/reagents/holder)
+			if(force > 3)
+				shock_ticks = 5
+				playsound(get_turf(holder.my_atom), 'sound/effects/bubbles_short.ogg', 50, 1)
+
+		on_reaction(datum/reagents/holder, created_volume)
+			reaction_icon_state = list("reaction_bubble-1", "reaction_bubble-2")
+			if(shock_ticks > 0) 
+				reaction_speed = 1
+				shock_ticks--
+				reaction_icon_state = list("reaction_sparkle-1", "reaction_sparkle-2")
+			else
+				reaction_speed = 0.5
+
+			//precipitate gross stuff, it's easy to MAKE a caffeine machine, but you want to clear out your equipment from time to time...
+			if (prob(50))
+				holder.add_reagent("bitters", created_volume*0.2, temp_new = holder.total_temperature, chemical_reaction = FALSE) //RAW coffee
```

Comment:
this making alcohol seems... odd?

Reply:
totally forgot that bitters had such a high ABV.
I didn't want pure yuck (it's weird clogging up a vessel with 100% purified food poisoning) and figured something tangentially related to coffee and gross would be fun. 
sewage as a stand-in could work here too. could also make exceptionally poorly made barrels of caffeine make you vomit as well as food poisoning.

i'll swap it out if that's thumbs-up worthy.

Reply:
Yeah that would probably work.
(and now a part of me is weeping at the idea of making espresso by mixing coffee with a drying agent lol)